core/domain/user_jobs_one_off.py

```diff
@@ -470,3 +470,120 @@ def map(user_settings_model):
     def reduce(key, values):
         """Implements the reduce function for this job."""
         yield (key, len(values))
+
+
+class CleanupUserSubscribersModelOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that cleans up UserSubscribersModel by removing user id if it is
+    present in subscriber ids.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [user_models.UserSubscribersModel]
+
+    @staticmethod
+    def map(item):
+        if not item.deleted and item.id in item.subscriber_ids:
+            item.subscriber_ids.remove(item.id)
+            item.put()
+            yield ('Cleaned up User Subscribers', 1)
+
+    @staticmethod
+    def reduce(key, values):
+        yield (key, len(values))
+
+
+class CleanupCollectionProgressModelOneOffJob(
+        jobs.BaseMapReduceOneOffJobManager):
+    """Job that cleans up CollectionProgressModel by:
+    1. Removing exploration ids which are not a part of the collection.
+    2. Creating CompletedActivitiesModel for completed explorations if it is
+    missing.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [user_models.CollectionProgressModel]
+
+    @staticmethod
+    def map(item):
+        if not item.deleted:
+            col_model = collection_models.CollectionModel.get_by_id(
+                item.collection_id)
+
+            collection_node_ids = [
+                node['exploration_id'] for node in (
+                    col_model.collection_contents['nodes'])]
+            exp_ids_to_remove = [
+                exp_id
+                for exp_id in item.completed_explorations if exp_id not in (
+                    collection_node_ids)]
+
+            if exp_ids_to_remove:
+                item.completed_explorations = [
+                    exp_id for exp_id in item.completed_explorations
+                    if exp_id not in exp_ids_to_remove]
+                item.put()
+                yield (
+                    'Invalid Exploartion IDs cleaned from '
```

Comment:
Incorrect spelling.

Maybe yield the ids of the exploration and the collection progress model in question, since this is an error case. Please fix this sort of thing for the other jobs in this and the other PRs.

Reply:
Done.